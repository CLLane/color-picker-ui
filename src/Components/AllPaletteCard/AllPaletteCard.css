.all-swatch__container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgb(245, 245, 245, 0.7);
  border-radius: 8px;
  margin: 10px;
  height: 50vh;
  width: 105vh;
  box-shadow: 4px 4px 10px black;
}

.swatch-name__paragraph {
  margin-bottom: 10px;
  color: black;
  font-weight: bold;
  font-size: 25px;
}

.inner-swatch__container {
  display: flex;
}

.swatch-expand__img {
  width: 50px;
  margin-top: 10px;
}