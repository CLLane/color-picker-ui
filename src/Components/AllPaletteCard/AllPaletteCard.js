import React from 'react';
import './AllPaletteCard.css';
import { Link } from 'react-router-dom';
import expandIcon from '../../Images/expandIcon.svg';

export const AllPaletteCard = ({ palette, grabPalette }) => {
  const colors = Object.values(palette).slice(3)
  const swatch = colors.map((hex, index) => {
    const divStyle = {
      background: hex,
      height: '15vh',
      width: '15vh',
      borderRadius: '8px',
      marginRight: '2.5px',
      marginLeft: '2.5px',
      marginTop: '3px',
    }
    return <div style={divStyle} key={index}></div>
  })


  return (
          <div className='all-swatch__container'>
            <p className='swatch-name__paragraph'>{palette.name}</p>
            <div className='inner-swatch__container'>
              {swatch}
            </div>
              <Link to="/">
                <img className='swatch-expand__img' src={expandIcon} alt='expand palette' onClick={() => grabPalette(palette)}/>
              </Link>
          </div>
  )
}

export default AllPaletteCard;