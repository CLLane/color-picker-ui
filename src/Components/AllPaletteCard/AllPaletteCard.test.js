import React from 'react';
import { shallow } from 'enzyme';
import AllPaletteCard from './AllPaletteCard';
import './AllPaletteCard.css';

describe('AllPalette Card', () => {
  let wrapper;
  let mockPalette;
  let mockTrashPalette;
  let mockGrabPalette;
  let mockShowPalette;
  beforeEach(() => {
    mockPalette = {
      id: 1, 
      name: 'Cool Colors',
      project_id: '12345',
      colorOne: '#FFFFFF',
      colorTwo: '#FFFFFF',
      colorThree: '#FFFFFF',
      colorFour: '#FFFFFF',
      colorFive: '#FFFFFF'
    };
    mockTrashPalette = jest.fn();
    mockGrabPalette = jest.fn();
    mockShowPalette = jest.fn();
    wrapper = shallow (<AllPaletteCard 
      palette={mockPalette}
      trashPalette={mockTrashPalette}
      grabPalette={mockGrabPalette}
      showPalette={mockShowPalette}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to pass the palette to the grabPalette function', () => {
    wrapper.find('.swatch-expand__img').simulate('click');
    expect(mockGrabPalette).toHaveBeenCalledWith(mockPalette);
  });
});