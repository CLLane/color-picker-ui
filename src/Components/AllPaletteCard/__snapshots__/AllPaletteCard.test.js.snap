// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPalette Card should match the snapshot 1`] = `
<div
  className="all-swatch__container"
>
  <p
    className="swatch-name__paragraph"
  >
    Cool Colors
  </p>
  <div
    className="inner-swatch__container"
  >
    <div
      key="0"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "15vh",
          "marginLeft": "2.5px",
          "marginRight": "2.5px",
          "marginTop": "3px",
          "width": "15vh",
        }
      }
    />
    <div
      key="1"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "15vh",
          "marginLeft": "2.5px",
          "marginRight": "2.5px",
          "marginTop": "3px",
          "width": "15vh",
        }
      }
    />
    <div
      key="2"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "15vh",
          "marginLeft": "2.5px",
          "marginRight": "2.5px",
          "marginTop": "3px",
          "width": "15vh",
        }
      }
    />
    <div
      key="3"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "15vh",
          "marginLeft": "2.5px",
          "marginRight": "2.5px",
          "marginTop": "3px",
          "width": "15vh",
        }
      }
    />
    <div
      key="4"
      style={
        Object {
          "background": "#FFFFFF",
          "borderRadius": "8px",
          "height": "15vh",
          "marginLeft": "2.5px",
          "marginRight": "2.5px",
          "marginTop": "3px",
          "width": "15vh",
        }
      }
    />
  </div>
  <Link
    to="/"
  >
    <img
      alt="expand palette"
      className="swatch-expand__img"
      onClick={[Function]}
      src="expandIcon.svg"
    />
  </Link>
</div>
`;
